File: Nx3CS/bin/test/src/nx3/PNoteHeadsRectsLyricsCalculator.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class PNoteHeadsRectsLyricsCalculator : global::haxe.lang.HxObject {
		public    PNoteHeadsRectsLyricsCalculator(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    PNoteHeadsRectsLyricsCalculator(global::nx3.PNote note, string text, object font){
			unchecked {
				#line 25 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
				global::nx3.PNoteHeadsRectsLyricsCalculator.__hx_ctor_nx3_PNoteHeadsRectsLyricsCalculator(this, note, text, font);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_PNoteHeadsRectsLyricsCalculator(global::nx3.PNoteHeadsRectsLyricsCalculator __temp_me107, global::nx3.PNote note, string text, object font){
			unchecked {
				#line 27 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
				__temp_me107.note = note;
				__temp_me107.text = text;
				__temp_me107.font = font;
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
				return new global::nx3.PNoteHeadsRectsLyricsCalculator(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
				return new global::nx3.PNoteHeadsRectsLyricsCalculator(((global::nx3.PNote) (arr[0]) ), global::haxe.lang.Runtime.toString(arr[1]), ((object) (arr[2]) ));
			}
			#line default
		}
		
		
		public  global::nx3.PNote note;
		
		public  string text;
		
		public  object font;
		
		public virtual   global::Array<object> getHeadsRects(){
			unchecked {
				#line 47 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
				global::haxe.Log.trace.__hx_invoke2_o(default(double), "non possible!", default(double), new global::haxe.lang.DynamicObject(new global::Array<int>(new int[]{302979532, 1547539107, 1648581351}), new global::Array<object>(new object[]{"getHeadsRects", "nx3.PNoteHeadsRectsLyricsCalculator", "PNoteHeadsRectsLyricsCalculator.hx"}), new global::Array<int>(new int[]{1981972957}), new global::Array<double>(new double[]{((double) (47) )})));
				#line 51 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
				return default(global::Array<object>);
			}
			#line default
		}
		
		
		public override   double __hx_setField_f(string field, int hash, double @value, bool handleProperties){
			unchecked {
				#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
				switch (hash){
					case 1136680399:
					{
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						this.font = ((object) (@value) );
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						return @value;
					}
					
					
					default:
					{
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						return base.__hx_setField_f(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
				switch (hash){
					case 1136680399:
					{
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						this.font = ((object) (@value) );
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						return @value;
					}
					
					
					case 1291439277:
					{
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						this.text = global::haxe.lang.Runtime.toString(@value);
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						return @value;
					}
					
					
					case 1225398258:
					{
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						this.note = ((global::nx3.PNote) (@value) );
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						return @value;
					}
					
					
					default:
					{
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
				switch (hash){
					case 325402290:
					{
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getHeadsRects") ), ((int) (325402290) ))) );
					}
					
					
					case 1136680399:
					{
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						return this.font;
					}
					
					
					case 1291439277:
					{
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						return this.text;
					}
					
					
					case 1225398258:
					{
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						return this.note;
					}
					
					
					default:
					{
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   double __hx_getField_f(string field, int hash, bool throwErrors, bool handleProperties){
			unchecked {
				#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
				switch (hash){
					case 1136680399:
					{
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						return ((double) (global::haxe.lang.Runtime.toDouble(this.font)) );
					}
					
					
					default:
					{
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						return base.__hx_getField_f(field, hash, throwErrors, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
				switch (hash){
					case 325402290:
					{
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						return this.getHeadsRects();
					}
					
					
					default:
					{
						#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
				baseArr.push("font");
				#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
				baseArr.push("text");
				#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
				baseArr.push("note");
				#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
				{
					#line 19 "F:\\nx3\\src\\nx3\\PNoteHeadsRectsLyricsCalculator.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
	}
}

File: Nx3Java/java/src/nx3/PNoteHeadsRectsLyricsCalculator.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PNoteHeadsRectsLyricsCalculator extends haxe.lang.HxObject
{
	public    PNoteHeadsRectsLyricsCalculator(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    PNoteHeadsRectsLyricsCalculator(nx3.PNote note, java.lang.String text, java.lang.Object font)
	{
		nx3.PNoteHeadsRectsLyricsCalculator.__hx_ctor_nx3_PNoteHeadsRectsLyricsCalculator(this, note, text, font);
	}
	
	
	public static   void __hx_ctor_nx3_PNoteHeadsRectsLyricsCalculator(nx3.PNoteHeadsRectsLyricsCalculator __temp_me110, nx3.PNote note, java.lang.String text, java.lang.Object font)
	{
		__temp_me110.note = note;
		__temp_me110.text = text;
		__temp_me110.font = font;
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.PNoteHeadsRectsLyricsCalculator(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.PNoteHeadsRectsLyricsCalculator(((nx3.PNote) (arr.__get(0)) ), haxe.lang.Runtime.toString(arr.__get(1)), ((java.lang.Object) (arr.__get(2)) ));
	}
	
	
	public  nx3.PNote note;
	
	public  java.lang.String text;
	
	public  java.lang.Object font;
	
	public   haxe.root.Array<nx3.geom.Rectangle> getHeadsRects()
	{
		haxe.Log.trace.__hx_invoke2_o(0.0, "non possible!", 0.0, new haxe.lang.DynamicObject(new haxe.root.Array<java.lang.String>(new java.lang.String[]{"className", "fileName", "methodName"}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{"nx3.PNoteHeadsRectsLyricsCalculator", "PNoteHeadsRectsLyricsCalculator.hx", "getHeadsRects"}), new haxe.root.Array<java.lang.String>(new java.lang.String[]{"lineNumber"}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{((java.lang.Object) (((double) (47) )) )})));
		return null;
	}
	
	
	@Override public   double __hx_setField_f(java.lang.String field, double value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef707 = true;
			switch (field.hashCode())
			{
				case 3148879:
				{
					if (field.equals("font")) 
					{
						__temp_executeDef707 = false;
						this.font = ((java.lang.Object) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef707) 
			{
				return super.__hx_setField_f(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef708 = true;
			switch (field.hashCode())
			{
				case 3148879:
				{
					if (field.equals("font")) 
					{
						__temp_executeDef708 = false;
						this.font = ((java.lang.Object) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 3387378:
				{
					if (field.equals("note")) 
					{
						__temp_executeDef708 = false;
						this.note = ((nx3.PNote) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 3556653:
				{
					if (field.equals("text")) 
					{
						__temp_executeDef708 = false;
						this.text = haxe.lang.Runtime.toString(value);
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef708) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef709 = true;
			switch (field.hashCode())
			{
				case 2051530802:
				{
					if (field.equals("getHeadsRects")) 
					{
						__temp_executeDef709 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getHeadsRects"))) );
					}
					
					break;
				}
				
				
				case 3387378:
				{
					if (field.equals("note")) 
					{
						__temp_executeDef709 = false;
						return this.note;
					}
					
					break;
				}
				
				
				case 3148879:
				{
					if (field.equals("font")) 
					{
						__temp_executeDef709 = false;
						return this.font;
					}
					
					break;
				}
				
				
				case 3556653:
				{
					if (field.equals("text")) 
					{
						__temp_executeDef709 = false;
						return this.text;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef709) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   double __hx_getField_f(java.lang.String field, boolean throwErrors, boolean handleProperties)
	{
		{
			boolean __temp_executeDef710 = true;
			switch (field.hashCode())
			{
				case 3148879:
				{
					if (field.equals("font")) 
					{
						__temp_executeDef710 = false;
						return ((double) (haxe.lang.Runtime.toDouble(this.font)) );
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef710) 
			{
				return super.__hx_getField_f(field, throwErrors, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef711 = true;
			switch (field.hashCode())
			{
				case 2051530802:
				{
					if (field.equals("getHeadsRects")) 
					{
						__temp_executeDef711 = false;
						return this.getHeadsRects();
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef711) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("font");
		baseArr.push("text");
		baseArr.push("note");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


